src/AppInstallerCommonCore/Public/winget/UserSettings.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "AppInstallerStrings.h"
+#include "winget/settings/Source.h"
+#include "winget/settings/Visual.h"
+#include <json.h>
+
+#include <filesystem>
+#include <vector>
+#include <string>
+
+namespace AppInstaller::Settings
+{
+    // The type of argument.
+    enum class UserSettingsType
+    {
+        // Settings files don't exist. A file is created on the first call to the settings command.
+        Default,
+        // Loaded settings.json
+        Standard,
+        // Loaded settings.json.backup
+        Backup,
+    };
+
+    // Representation of the parsed settings file.
+    struct UserSettings
+    {
+        static UserSettings& Instance()
+        {
+            // If we go multi-threaded secure this.
+            static UserSettings userSettings;
+            return userSettings;
+        }
+
+        UserSettings(const UserSettings&) = delete;
+        UserSettings& operator=(const UserSettings&) = delete;
+
+        UserSettingsType GetType() const { return m_type; }
+        std::vector<std::string> GetWarnings() const { return m_warnings; }
+
+        void Reload();
+        void CreateFileIfNeeded();
+        void CreateBackup();
```

Comment:
These feel like they should be static.

Reply:
not needed anymore

---
In reply to: [442426732](https://github.com/microsoft/winget-cli/pull/436#discussion_r442426732) [](ancestors = 442426732)